packages/ui/src/signInWithGoogleButton.tsx

```diff
@@ -0,0 +1,132 @@
+/* eslint-disable react-hooks/rules-of-hooks */
+'use client';
+
+import { useEffect, useState } from 'react';
+import GoogleIcon from '@mui/icons-material/Google';
+import { Avatar, Button, Menu, MenuItem, Typography } from '@mui/material';
+import { signOut } from 'firebase/auth';
+import firebase from 'firebase/compat/app';
+
+import 'firebase/compat/auth';
+
+const firebaseConfig = {
+	apiKey: process.env.NEXT_PUBLIC_FIREBASE_API_KEY,
+	authDomain: process.env.NEXT_PUBLIC_FIREBASE_AUTH_DOMAIN,
+	projectId: process.env.NEXT_PUBLIC_FIREBASE_PROJECT_ID,
+	storageBucket: process.env.NEXT_PUBLIC_FIREBASE_STORAGE_BUCKET,
+	messagingSenderId: process.env.NEXT_PUBLIC_FIREBASE_MESSAGING_SENDER_ID,
+	appId: process.env.NEXT_PUBLIC_FIREBASE_APP_ID,
+};
```

Comment:
this shouldn't be here, put it under e.g `utils/config.ts` then other part of the source code an access.